vita3k/renderer/src/gl/draw.cpp

```diff
@@ -128,7 +128,7 @@ void draw(GLState &renderer, GLContext &context, const FeatureState &features, S
     set_uniform_buffer(context, true, SCE_GXM_REAL_MAX_UNIFORM_BUFFER, sizeof(GXMRenderVertUniformBlock), &vert_ublock, false);
 
     GXMRenderFragUniformBlock frag_ublock;
-    const bool both_side_fragment_program_disabled = context.record.front_side_fragment_program_mode && (context.record.front_side_fragment_program_mode || context.record.back_side_fragment_program_mode);
+    const bool both_side_fragment_program_disabled = context.record.front_side_fragment_program_mode && context.record.back_side_fragment_program_mode;
```

Comment:
Nice one. Can you change the name of variable to both_side_fragment_program_enabled?

Reply:
No. 
`enum SceGxmFragmentProgramMode {
    SCE_GXM_FRAGMENT_PROGRAM_DISABLED = 0x00200000u,
    SCE_GXM_FRAGMENT_PROGRAM_ENABLED = 0x00000000u
};`
so True mean disabled.
Or is there also some logic bug?

Reply:
https://github.com/Vita3K/Vita3K/blob/9cb2063d50ae5fe0603594ea6122069cf8d67510/vita3k/shader/src/spirv_recompiler.cpp#L1271

We disable it when disabled != 0.0f
 
should be context.record.front_side_fragment_program_mode == SCE_GXM_FRAGMENT_PROGRAM_DISABLED && context.record.back_side_fragment_program_mode == SCE_GXM_FRAGMENT_PROGRAM_DISABLED ;

Reply:
What about this block?
https://github.com/Vita3K/Vita3K/blob/399fe1766f0f985d440813def0ab2aef53ea12ea/vita3k/renderer/src/gl/draw.cpp#L194-L196

Reply:
should it be  if (both_side_fragment_program_enabled) { 
or  if (!both_side_fragment_program_enabled) {

Reply:
No that line should be fine. When fragment program is disabled, we should not write the color that was given by fragment program but should still write depth. But, we're using discard to implement this behavior so depth is also thrown away. We're temporarily masking color write when both side is disabled, but doesn't discard pixels so that depth buffer is written. It's syncing blending state to restore the tweaked color mask.

Reply:
A cleaner way to approach this thing is don't use discard but write the previous color. Then, we don't need this color write trickery. This fix needs a bit more code tho. Let's only fix the logic error of both_side_fragment_program_disabled in this pr.

Reply:
both_side_fragment_program_disabled was the right name btw. Sorry for confusion.